libsolidity/ast/AST.h

```diff
@@ -587,10 +587,19 @@ class ContractDefinition: public Declaration, public StructurallyDocumented, pub
 class IdentifierPath: public ASTNode
 {
 public:
-	IdentifierPath(int64_t _id, SourceLocation const& _location, std::vector<ASTString> _path):
-		ASTNode(_id, _location), m_path(std::move(_path)) {}
+	IdentifierPath(
+		int64_t _id,
+		SourceLocation const& _location,
+		std::vector<ASTString> _path,
+		std::vector<SourceLocation> _pathLocations
```

Comment:
I assume that the path locations are the source locations of the individual elements, i.e. in `a.b.c.d`, the location of the `a`, then of the `b`, then of the `c` and so on. In contrast to other ast elements, these are not nested. So the other interpretation could be that it is the location of `a`, then of `a.b` then of `a.b.c` and so on.
Please document which one is the case.